lib/src/main/java/edu/stanford/futuredata/macrobase/analysis/summary/aplinear/APLExplanationResult.java

```diff
@@ -30,41 +27,58 @@ public APLExplanationResult(
         this.metrics = metrics;
     }
 
-    @Override
-    public String toString() {
-        return "a="+Arrays.toString(matcher.values)+":ag="+Arrays.toString(aggregates)+":mt="+Arrays.toString(metrics);
-    }
-
-    public String prettyPrint(
-            AttributeEncoder encoder,
-            List<String> aggregateNames
-    ) {
+    private Map<String, String> prettyPrintMatch(AttributeEncoder encoder) {
         Set<Integer> values = matcher.getSet();
+        Map<String, String> match = new HashMap<>();
 
-        StringJoiner matchJoiner = new StringJoiner(", ");
         for (int k : values) {
-            matchJoiner.add(encoder.decodeColumnName(k)+"="+encoder.decodeValue(k));
+            match.put(encoder.decodeColumnName(k), encoder.decodeValue(k));
         }
+        return match;
+    }
+
+    private Map<String, String> prettyPrintMetric() {
+        Map<String, String> metric = new HashMap<>();
 
-        StringJoiner metricJoiner = new StringJoiner(", ");
         for (int i = 0; i < metricTypes.length; i++) {
-            metricJoiner.add(
-                    String.format("%s: %.3f", metricTypes[i].name(), metrics[i])
-            );
+            metric.put(metricTypes[i].name(), String.format("%.3f", metrics[i]));
         }
+        return metric;
+    }
+
+    private Map<String, String> prettyPrintAggregate(List<String> aggregateNames) {
+        Map<String, String> aggregate = new HashMap<>();
 
-        StringJoiner aggregateJoiner = new StringJoiner(", ");
         for (int i = 0; i < aggregates.length; i++) {
-            aggregateJoiner.add(
-                    String.format("%s: %.3f", aggregateNames.get(i), aggregates[i])
-            );
+            aggregate.put(aggregateNames.get(i), String.format("%.3f", aggregates[i]));
         }
+        return aggregate;
+    }
+
+    @Override
+    public String toString() {
+        return "a="+Arrays.toString(matcher.values)+":ag="+Arrays.toString(aggregates)+":mt="+Arrays.toString(metrics);
+    }
+
+    public Map<String, Map<String, String>> jsonPrint(AttributeEncoder encoder,
+                                  List<String> aggregateNames) {
+        return new HashMap<String, Map<String, String>>() {{
+            put("matcher", prettyPrintMatch(encoder));
+            put("metric", prettyPrintMetric());
+            put("aggregate", prettyPrintAggregate(aggregateNames));
 
+        }};
+    }
+
+    public String prettyPrint(
```

Comment:
Should make prettyprint a bit more human readable, e.g. remove braces and add a label for what the metric and match portions are, maybe something like:
```
metrics: global_ratio=52.850, support=1.000
matches: location=CAN, version=v3
aggregates: Outliers=6.000, Count=20.000
```